File: Practica General/POO/Ejercicio 1/Persona.java
public class Persona {
    private String nombre;
    private int edad;
    private String ciudad;

    public Persona(String nombre, int edad, String ciudad) {
        this.nombre = nombre;
        this.edad = edad;
        this.ciudad = ciudad;
    }

    public void saludar() {
        System.out.println("Hola, soy " + nombre + " de " + ciudad);
    }

    public boolean esMayorEdad() {
        return edad >= 18;
    }

    public static void main(String[] args) {
        Persona p1 = new Persona("Ana", 22, "La Paz");
        Persona p2 = new Persona("Luis", 17, "Cochabamba");
        Persona p3 = new Persona("Marta", 30, "Santa Cruz");

        Persona[] personas = {p1, p2, p3};

        for (Persona p : personas) {
            p.saludar();
            System.out.println("Mayor de edad: " + (p.esMayorEdad() ? "Sí" : "No"));
            System.out.println("---");
        }
    }
}

File: Practica General/POO/Ejercicio 1/persona.py
class Persona:
    def __init__(self, nombre, edad, ciudad):
        self.nombre = nombre
        self.edad = edad
        self.ciudad = ciudad

    def saludar(self):
        print(f"Hola, soy {self.nombre} de {self.ciudad}")

    def es_mayor_edad(self):
        return self.edad >= 18


# Prueba
persona1 = Persona("Ana", 22, "La Paz")
persona2 = Persona("Luis", 17, "Cochabamba")
persona3 = Persona("Marta", 30, "Santa Cruz")

personas = [persona1, persona2, persona3]

for p in personas:
    p.saludar()
    print("Mayor de edad:", "Sí" if p.es_mayor_edad() else "No")
    print("---")
